config/configgrpc/configgrpc.go

```diff
@@ -85,24 +133,68 @@ func GrpcSettingsToDialOptions(settings GRPCClientSettings) ([]grpc.DialOption,
 	}
 	opts = append(opts, tlsDialOption)
 
-	if settings.KeepaliveParameters != nil {
+	if settings.Keepalive != nil {
 		keepAliveOption := grpc.WithKeepaliveParams(keepalive.ClientParameters{
-			Time:                settings.KeepaliveParameters.Time,
-			Timeout:             settings.KeepaliveParameters.Timeout,
-			PermitWithoutStream: settings.KeepaliveParameters.PermitWithoutStream,
+			Time:                settings.Keepalive.Time,
+			Timeout:             settings.Keepalive.Timeout,
+			PermitWithoutStream: settings.Keepalive.PermitWithoutStream,
 		})
 		opts = append(opts, keepAliveOption)
 	}
 
 	return opts, nil
 }
 
-var (
-	// Map of opentelemetry compression types to grpc registered compression types
-	grpcCompressionKeyMap = map[string]string{
-		CompressionGzip: gzip.Name,
+// ToDialOptions maps configgrpc.GRPCServerSettings to a slice of dial options for gRPC
+func (gss *GRPCServerSettings) ToDialOptions() ([]grpc.ServerOption, error) {
```

Comment:
Should this function be called `ToServerOptions`?

Reply:
Done.